viewer/src/dataModels/pointCloud/internal/PointCloudFactory.ts

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+// @ts-ignore
+import * as Potree from '@cognite/potree-core';
+import { PointCloudMetadata } from '@/dataModels/pointCloud/public/PointCloudMetadata';
+import { PotreeNodeWrapper } from './PotreeNodeWrapper';
+import { toThreeMatrix4 } from '@/utilities/utilities';
+import { HttpHeadersProvider } from '@/utilities/networking/HttpHeadersProvider';
+
+export class PointCloudFactory {
+  private readonly _httpHeadersProvider: HttpHeadersProvider;
+
+  constructor(httpHeadersProvider: HttpHeadersProvider) {
+    this._httpHeadersProvider = httpHeadersProvider;
+  }
+
+  createModel(modelMetadata: PointCloudMetadata): PotreeNodeWrapper {
+    this.initializePointCloudXhrRequestHeaders();
+    const { blobUrl, modelTransformation, scene } = modelMetadata;
+    const geometry = new Potree.PointCloudEptGeometry(blobUrl + '/', scene);
+    const x = geometry.offset.x;
+    const y = geometry.offset.y;
+    const z = geometry.offset.z;
+    const root = new Potree.PointCloudEptGeometryNode(geometry, geometry.boundingBox, 0, x, y, z);
+    geometry.root = root;
+    geometry.root.load();
+
+    const octtree = new Potree.PointCloudOctree(geometry);
+    octtree.name = 'PointCloudOctree'; // TODO: 25-05-2020 j-bjorne Should we just name it like this?
```

Comment:
Add `blobUrl` to identifier to make it possible to distinguish multiple instances

Reply:
Done